test/net/sourceforge/kolmafia/textui/ParserTest.java

```diff
@@ -486,7 +486,7 @@
 			{
 				"Primitive assignment to aggregate",
 				"int[4] x = 1;",
-				"Cannot store int in x of type int [4]",
+				"Cannot store int in a variable of type int [4]",
```

Comment:
Do you think it would be more helpful to say "Cannot store int in variable x of type int[4]"?

(Namely, call out _which_ variable is getting a bad assignment)

Maybe it's moot with line number + column info.

(Other than this, the PR looks reasonable.)

Reply:
The isolation of the initialization into a method means the name of the variable is now out of reach...

Reply:
That seems like a situation where you'd catch the exception at the callsite, then rethrow it with more context.

It's basically the same rationale as extending "Unexpected = at line X" with " in file script.ash". (Not that Parser throws exceptions that look like that today... but just as an example.)

So, we could envision something like...

"Cannot store int in type int[4]" + " for variable x".

(It's not super important for this PR, as it will look weird initially with () representing the lack of a file. But you can fix that with the same pattern by shunting the file annotation into the corresponding parse() call.)